### PR TITLE
Added Isbsvc service name to the pipeline label

### PR DESCRIPTION
**Pull request by shubhamdixit863**
Yes

Fixes  https://github.com/numaproj/numaplane/issues/98

### Modifications

Modified PipelineRolloutReconciler reconcile method to Add ISBSVC name as a label to the Pipeline

### Verification

Yet to test

### CODE REVIEW

File: internal/controller/pipelinerollout_controller.go [hunk 1]
```diff
@@ -280,14 +282,24 @@ func (r *PipelineRolloutReconciler) reconcile(
 		controllerutil.AddFinalizer(pipelineRollout, finalizerName)
 	}
 
+	var pipelineSpec struct {
+		InterStepBufferServiceName string `json:"interStepBufferServiceName"`
+	}
+	if err := json.Unmarshal(pipelineRollout.Spec.Pipeline.Raw, &pipelineSpec); err != nil {
+		return false, fmt.Errorf("failed to unmarshal pipeline spec: %v", err)
+	}
+	labelKey := "isbsvc-name"
```
Comment (afugazzotto) [inline internal/controller/pipelinerollout_controller.go:291]: Could you make a function that given a PipelineRollout returns a map of labels? In this case the map will only have the isbs-name but we could add more logic to the function to add more labels as needed. Then, you can call that func to set the labels of the newPipelineDef object.

File: internal/controller/pipelinerollout_controller.go [hunk 2]
```diff
@@ -280,14 +282,24 @@ func (r *PipelineRolloutReconciler) reconcile(
 		controllerutil.AddFinalizer(pipelineRollout, finalizerName)
 	}
 
+	var pipelineSpec struct {
+		InterStepBufferServiceName string `json:"interStepBufferServiceName"`
```
Comment (juliev0) [inline internal/controller/pipelinerollout_controller.go:286]: if this field isn't found, we can set the Label to "default"

File: internal/controller/pipelinerollout_controller.go [hunk 3]
```diff
@@ -504,6 +511,23 @@ func applyPipelineSpec(
 	return nil
 }
 
+func pipeLinLabels(pipelineRollout *apiv1.PipelineRollout) (map[string]string, error) {
```
Comment (juliev0) [inline internal/controller/pipelinerollout_controller.go:514]: can we rename to `pipelineLabels()`?

File: internal/controller/pipelinerollout_controller_test.go [hunk 4]
```diff
@@ -288,3 +290,49 @@ var _ = Describe("PipelineRollout Controller", Ordered, func() {
 		})
 	})
 })
+
+func TestPipeLinLabels(t *testing.T) {
```
Comment (juliev0) [inline internal/controller/pipelinerollout_controller_test.go:294]: thank you for adding this :)  although, going forward, don't worry about adding a test for pretty simple logic like this

**Review by juliev0 — APPROVED** [review]

**Comment by juliev0** [comment]
is this ready for review @shubhamdixit863 ?

**Comment by shubhamdixit863** [comment]
> is this ready for review @shubhamdixit863 ?

Yes all changes done

**Review by afugazzotto — APPROVED** [review]

**Pull request merged**